packages/driver/src/cy/multi-domain/index.ts

```diff
@@ -144,56 +128,142 @@ export function addCommands (Commands, Cypress: Cypress.Cypress, cy: Cypress.cy,
         Cypress.action('cy:enqueue:command', attrs)
       }
 
-      const updateCommand = (details) => {
-        if (details.logAdded) {
-          const attrs = details.logAdded
+      const updateCommand = ({ id, end }) => {
+        if (end) {
+          const command = commands[id]
+
+          if (command) {
+            delete commands[id]
+            command.deferred.resolve()
+          }
+        }
+      }
 
-          attrs.consoleProps = () => details.logAdded.consoleProps
-          attrs.renderProps = () => details.logAdded.renderProps
+      const onLogAdded = (logAdded) => {
+        const attrs = logAdded
 
-          const log = Cypress.log(attrs)
+        attrs.consoleProps = () => logAdded.consoleProps
+        attrs.renderProps = () => logAdded.renderProps
 
-          logs[log.get('id')] = log
+        const log = Cypress.log(attrs)
 
-          return
+        // if the log needs to stream updates, defer its result to make sure all streamed updates come in
+        if (!attrs.ended) {
+          logs[log.get('id')] = {
+            log,
+            deferred: createDeferred(),
+          }
         }
+      }
 
-        if (details.logChanged) {
-          const log = logs[details.logChanged.id]
+      const onLogChanged = (logChanged) => {
+        const { deferred, log } = logs[logChanged.id]
 
-          if (log) {
-            log.set(details.logChanged)
+        // NOTE: Cypress.LogConfig only contains partial types of what exists on the log attributes, missing a lot of 'private' properties
```

Comment:
If it would help to create additional "internal" types, the `driver` already has some types for `Cypress.LogConfig` here that can be extended: https://github.com/cypress-io/cypress/blob/1f28650d684c0b35265983edb92d58db325be819/packages/driver/types/internal-types.d.ts#L49-L68

Reply:
I didn't actually notice the name of the interface file 😂 . I can go ahead the ones we are using here as well as some others!

Reply:
@flotwig @chrisbreiding I tried to enhance the log types on 4429a9742b1fbf7b3b4c2a21dd2a12889f942875. Some of it is a bit out of scope of this PR, but wanted to get y'alls opinion on it. Tried to get some types in there that aren't currently documented, but are used elsewhere in the codebase